### PR TITLE
getprop: add page

### PR DESCRIPTION
**Pull request by marchersimon**
Part of #4934.

### CODE REVIEW

File: pages/common/adb-shell-getprop.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+`adb shell getprop {{ro.vendor.product.model}}`
+
+- Print the OEM unlock status:
+
+`adb shell getprop {{ro.oem_unlock_supported}}`
+
+- Print the MAC address of the Android's WiFi card:
+
+`adb shell getprop ro.boot.wifimacaddr`
```
Comment (bl-ue) [inline pages/common/adb-shell-getprop.md:28]:
```suggestion
`adb shell getprop {{ro.boot.wifimacaddr}}`
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
Looks good, @marchersimon, however, I'd add one more thing: in the second position, add a example `adb shell getprop {{prop}}`, just to demonstrate that, you know?

**Review by navarroaxel — CHANGES_REQUESTED** [review]
Hey @marchersimon, thanks for your submission. But, like [`logcat.md`](https://github.com/tldr-pages/tldr/blob/master/pages/linux/logcat.md), I think these pages should be as a top level command: `getprop.md` and not a command inside `adb shell`.

**Comment by marchersimon** [comment]
But I can't execute `getprop` as a top level command. And `logcat` is only a top level command, because it isn't part of `adb shell`.

**Comment by navarroaxel** [comment]
> But I can't execute `getprop` as a top level command. And `logcat` is only a top level command, because it isn't part of `adb shell`.

You can start a shell into the Android device with `adb shell` and you can run it as a top level command, I can't remember exactly but I think we talk about add these kind of commands in `common/`

The command description in `logcat` says `Native Android CLI tool.` to let the reader know `logcat` is a command for the Android platform.

Let me invoke @sbrl

**Comment by marchersimon** [comment]
Okay, sure, I'll change it. But then we should add that the commands can only be executed inside `adb shell` to avoid confusion.

**Comment by sbrl** [comment]
I'm not sure how this command works, but I would generally say that we want to file a command as one that can be executed in an OS-native shell. For example, if I could open a Bash terminal on Linux or a Windows Command Prompt and execute `getprop` and have it work as expected, then we should file it under `getprop`.

On the other hand, if the command only works an ab Android Debugging Shell, then it should be filed under something that can be executed from an os-native shell. For example, if `adb shell <command>` can be used to execute android debugging shell commands directly, it should be filed under that - e.g. `adb shell getprop`.

In short, the question is basically "how can I execute this from an OS-native shell"?

File: pages/android/getprop.md [hunk 2]
```diff
@@ -1,32 +1,32 @@
-# adb shell getprop
+# getprop
 
 > Show information about adb devices and emulators.
 > More information: <https://adbinstaller.com/commands>.
```
Comment (sbrl) [inline pages/android/getprop.md:4]: This URL really isn't very specific. Is there something better we can reference?

Comment (bl-ue) [reply pages/android/getprop.md:4]: @sbrl https://github.com/tldr-pages/tldr/pull/5457#discussion_r595099178

Comment (sbrl) [reply pages/android/getprop.md:4]: As I mentioned in the other PR, I'm not convinced of the usefulness of the generic link here.

Comment (marchersimon) [reply pages/android/getprop.md:4]: I found [https://adbshell.com/commands/adb-shell-getprop](https://adbshell.com/commands/adb-shell-getprop), but the commands all start with `adb shell`.

File: pages/android/getprop.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# getprop
+
+> Show information about adb devices and emulators.
+> More information: <https://adbinstaller.com/commands>.
+
+- Print information about the Android system properties:
```
Comment (navarroaxel) [inline pages/android/getprop.md:6]: I prefer `display` as verb.
```suggestion
- Display information about the Android system properties:
```

**Review by navarroaxel — APPROVED** [review]

**Review by bl-ue — APPROVED** [review]
We just need the url sorted out.  @sbrl what do you think about https://github.com/tldr-pages/tldr/pull/5456/files#r602786840?

File: pages/android/getprop.md (hunk 2 above)
Comment (sbrl) [reply pages/android/getprop.md:4]: Hrm, not a big fan of that either really sorry.

**Comment by sbrl** [comment]
@bl-ue nope, not a fan of that either really.

Neither option really gets across that you can execute the commands on the native device. `adbinstaller.com` in particular seems to indicate that you need to somehow "install the adb onto the device", which doesn't seem right to me.

**Comment by marchersimon** [comment]
Adbinstaller just provides a program to execute android commands via a pc. But the command is natively available on android.

**Comment by marchersimon** [comment]
<https://cs.android.com/android/platform/superproject/+/master:system/core/toolbox/getprop.cpp> looks good.

**Comment by bl-ue** [comment]
That url looks fine to me. [Here](https://github.com/tldr-pages/tldr/blob/master/pages/linux/ksvgtopng5.md) we're linking to source files as well.

File: pages/android/getprop.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# getprop
+
+> Show information about adb devices and emulators.
+> More information: <https://cs.android.com/android/platform/superproject/+/master:system/core/toolbox/getprop.cpp>.
+
+- Display information about the Android system properties:
+
+`getprop`
+
+- Print information about a specific property:
```
Comment (navarroaxel) [inline pages/android/getprop.md:10]:
```suggestion
- Display information about a specific property:
```

File: pages/android/getprop.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+`getprop {{prop}}`
+
+- Print the SDK API level:
```
Comment (navarroaxel) [inline pages/android/getprop.md:14]:
```suggestion
- Display the SDK API level:
```

File: pages/android/getprop.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+`getprop {{ro.build.version.sdk}}`
+
+- Print the Android version:
```
Comment (navarroaxel) [inline pages/android/getprop.md:18]:
```suggestion
- Display the Android version:
```

File: pages/android/getprop.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+`getprop {{ro.build.version.release}}`
+
+- Print the Android device model:
```
Comment (navarroaxel) [inline pages/android/getprop.md:22]:
```suggestion
- Display the Android device model:
```

File: pages/android/getprop.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+`getprop {{ro.vendor.product.model}}`
+
+- Print the OEM unlock status:
```
Comment (navarroaxel) [inline pages/android/getprop.md:26]:
```suggestion
- Display the OEM unlock status:
```

File: pages/android/getprop.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+`getprop {{ro.oem_unlock_supported}}`
+
+- Print the MAC address of the Android's WiFi card:
```
Comment (navarroaxel) [inline pages/android/getprop.md:30]:
```suggestion
- Display the MAC address of the Android's WiFi card:
```

**Review by navarroaxel — APPROVED** [review]

**Review by sbrl — APPROVED** [review]
Ok, this looks fine to me.

We really shouldn't make a habit of linking to the source code at all, but it's better than linking to what's essentially a tldr page or an unrelated homepage.

I'm going to merge this, but if anyone has any further comments on the matter please do open another PR. Linking to the source code is still far from desirable, but Android docs? what docs?

**Comment by bl-ue** [comment]
@marchersimon and @Rudloff have stated in multiple PRs that the Android tools are poorly documented... 🤷🏻

**Comment by Rudloff** [comment]
It looks like Ubuntu has [a man page for getprop](http://manpages.ubuntu.com/manpages/bionic/en/man1/getprop.1.html).

**Comment by bl-ue** [comment]
Ah yes and manned.org as well: https://manned.org/getprop

**Comment by sbrl** [comment]
Ooooh awesome @Rudloff, @bl-ue! Can someone open another PR to update this page then? That looks _much_ more preferable. I would suggest in this instance we link to manned.org, because `getprop` isn't an Ubuntu-specific command as far as I know.
